lib/Runtime/Library/TypedArray.cpp

```diff
@@ -606,15 +606,39 @@ namespace Js
             object;
     };
 
-    BOOL TypedArrayBase::HasProperty(PropertyId propertyId)
+    BOOL TypedArrayBase::HasOwnProperty(PropertyId propertyId)
+    {
+        return HasPropertyEx(propertyId) == Property_Found;
+    }
+
+    inline BOOL TypedArrayBase::CanonicalNumericIndexString(PropertyId propertyId, ScriptContext *scriptContext)
+    {
+        PropertyString *propertyString = scriptContext->GetPropertyString(propertyId);
```

Comment:
I agree. Based on perf numbers, didn't see a need for some kind of caching.

---
In reply to: [105254349](https://github.com/Microsoft/ChakraCore/pull/2640#discussion_r105254349) [](ancestors = 105254349)